File: code-examples/cpp/connection/socks.cpp
#include <mongocxx/client.hpp>
#include <mongocxx/stdx.hpp>
#include <mongocxx/uri.hpp>
#include <mongocxx/instance.hpp>
#include <bsoncxx/json.hpp>

int main() {
    mongocxx::instance inst{};
    mongocxx::uri uri("<connection string uri>");
    std::string host = "<host>";
    std::uint16_t port = 1080;
    std::string username = "<username>";
    std::string password = "<password>";
    
    uri.default_option("proxy=socks5://" + username + ":" + password + "@" + host + ":" + std::to_string(port));

    mongocxx::client conn{uri};

    bsoncxx::stdx::optional<bsoncxx::document::value> maybe_result =
        conn["myDB"]["myColl"].find_one({});

    if(maybe_result) {
        std::cout << bsoncxx::to_json(*maybe_result) << "\n";
    }

    return 0;
}

File: code-examples/pymongo/connection/socks.py
import pymongo
import socks
import socket
from getpass import getpass

# Replace the placeholders with your SOCKS5 proxy server details
socks.set_default_proxy(socks.PROXY_TYPE_SOCKS5, "<host>", 1080, 
                        username="<username>", password="<password>")
socket.socket = socks.socksocket

# Replace the placeholder with your connection string
client = pymongo.MongoClient("<connection string uri>")

try:
    # Connecting to a database
    db = client['myDB']
    
    # Selecting a collection
    my_coll = db['myColl']
    
    # Fetching the first document
    first_doc = my_coll.find_one()

    # Print the retrieved document
    print(first_doc)

except Exception as e:
    print(e)

finally:
    # Close the connection after the operation completes
    client.close()